extensions/panache/rest-data-panache/deployment/src/main/java/io/quarkus/rest/data/panache/deployment/methods/UserMethodsWithTransactionalImplementor.java

```diff
@@ -0,0 +1,62 @@
+package io.quarkus.rest.data.panache.deployment.methods;
+
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.DotName;
+
+import io.quarkus.deployment.Capabilities;
+import io.quarkus.deployment.Capability;
+import io.quarkus.gizmo.ClassCreator;
+import io.quarkus.gizmo.FieldDescriptor;
+import io.quarkus.gizmo.MethodCreator;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+import io.quarkus.rest.data.panache.deployment.ResourceMetadata;
+import io.quarkus.rest.data.panache.deployment.properties.ResourceProperties;
+
+/**
+ * Propagate the user methods annotated with `@Transactional`.
+ * This implementor is only used if Hibernate ORM is present.
+ */
+public final class UserMethodsWithTransactionalImplementor implements MethodImplementor {
+
+    public static final DotName TRANSACTIONAL = DotName.createSimple("jakarta.transaction.Transactional");
```

Comment:
Since from CDI perspective this is an interceptor binding like any other, shouldn't we look into treating all interceptor bindings the same as opposed to only doing it for `@Transactional`?
Furthermore interceptor binding [can itself declare more interceptor bindings](https://jakarta.ee/specifications/interceptors/2.0/interceptors-spec-2.0.html#interceptor_binding_types_with_additional_interceptor_bindings) (so `@Transactionl` might be inside another binding annotation) which this solution won't cover.
CC @mkouba

Reply:
this seems reasonable to me. Will work on it asap.

Reply:
Unfortunately, I can't consume `ContextRegistrationPhaseBuildItem` because it causes a cycle:

```java
@BuildStep
    void implementResources(CombinedIndexBuildItem index,
            List<RestDataResourceBuildItem> resourceBuildItems,
            List<ResourcePropertiesBuildItem> resourcePropertiesBuildItems,
            List<BuildTimeConditionBuildItem> buildTimeConditions,
            Capabilities capabilities,
            ContextRegistrationPhaseBuildItem contextRegistration, <1>
            BuildProducer<GeneratedBeanBuildItem> resteasyClassicImplementationsProducer, <2>
            BuildProducer<GeneratedJaxRsResourceBuildItem> resteasyReactiveImplementationsProducer) {
```

Since we're generating a bean in <2>, it causes that <1> is produced again, and hence the cycle.

Reply:
In that case, you can consume the `BeanArchiveIndexBuildItem#getImmutableIndex()` and mimic the logic used in `io.quarkus.arc.processor.BeanDeployment.findInterceptorBindings()`. You can ignore transitive interceptor bindings for now and add a TODO to the code.

Reply:
I'm considering copying all the user methods (regardless it uses `@Transactional` or any other annotation for an interceptor binding).
Therefore, the generated resource will include the user methods (and their annotations) and these methods will make the actual invocation to the user method.

Reply:
see https://github.com/quarkusio/quarkus/pull/35165, let's continue the discussion there. Waiting for you both for the review!